metaflow/plugins/kfp/kfp.py

```diff
@@ -998,23 +748,67 @@ def build_kfp_dag(
                     METAFLOW_USER=METAFLOW_USER,
                 )
                 metaflow_run_id = f"kfp-{dsl.RUN_ID_PLACEHOLDER}"
-                step_op_args = dict(
-                    cmd_template=kfp_component.cmd_template,
-                    metaflow_run_id=metaflow_run_id,
-                    metaflow_configs=metaflow_configs,
-                    passed_in_split_indexes=passed_in_split_indexes,
-                    preceding_component_inputs=preceding_component_inputs,
-                    preceding_component_outputs=kfp_component.preceding_component_outputs,
-                    flow_parameters_json=flow_parameters_json
-                    if node.name == "start"
-                    else None,
+
+                command = [
```

Comment:
Is there any way we can improve the output formatting of this command? I'm thinking an easy first step would be to reorder the arguments to be in alphabetical order. Then maybe we can take back a couple of bytes by formatting the output like:

```bash

cd /opt/zillow/aip_workflow_example \
&& python -m
    metaflow.plugins.kfp.kfp_step_function \
    --metaflow_run_id kfp-3c260f11-821b-4178-890b-f3e50818183c \
    --metaflow_configs "{\"METAFLOW_DATASTORE_SYSROOT_S3\": null, \"METAFLOW_USER\": null}" \
    --cd_into_metaflow_package_cmd " && cd /opt/zillow/aip_workflow_example" \
    --clean_volume_cmd "true" \
    --task_id kfp1 \
    --task_id_template "kfp1" \
    --step_name start \
    --flow_name ExampleFlow \
    --tags "[]" \
    --environment_type local \
    --logger_type nullSidecarLogger \
    --monitor_type nullSidecarMonitor \
    --user_code_retries 3 \
    --workflow_name aip-workflow-example-train-9zwr5 \
    --script_name example_flow.py \
    --passed_in_split_indexes= \
    --preceding_component_inputs="[]" \
    --preceding_component_outputs="[]" \
    --flow_parameters_json='{"data_path":"s3://serve-datalake-zillowgroup/zillow/ai-platform/aip-example/dev/data"}';
c=$?;
exit $c
```

That'd make the `Workflow` and `Pod` spec that bit easier to read/debug. Thoughts?

PS. Happy to defer to a follow up ticket this the scope/priority of this work if needs be!

Reply:
Hey Alex, I created [AIP-5406](https://zbrt.atl.zillow.net/browse/AIP-5406) to track this.